CondFormats/EcalObjects/interface/EcalMultifitConditionsSoA.h

```diff
@@ -0,0 +1,58 @@
+#ifndef CondFormats_EcalObjects_EcalMultifitConditionsSoA_h
+#define CondFormats_EcalObjects_EcalMultifitConditionsSoA_h
+
+#include <array>
+#include <Eigen/Dense>
+#include "DataFormats/SoATemplate/interface/SoACommon.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+#include "DataFormats/SoATemplate/interface/SoAView.h"
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+#include "CondFormats/EcalObjects/interface/EcalPulseShapes.h"
+
+using PulseShapeArray = std::array<float, EcalPulseShape::TEMPLATESAMPLES>;
+using SampleCorrelationArray = std::array<double, ecalPh1::sampleSize>;
+
+using CovarianceMatrix = Eigen::Matrix<float, EcalPulseShape::TEMPLATESAMPLES, EcalPulseShape::TEMPLATESAMPLES>;
+
+constexpr size_t N_TIMEBIASCORRECTIONS_BINS_EB = 71;
```

Comment:
Are these number of bins for the time bias corrections fixed forever? Shouldn't they be taken from some centralized location, if so?

Reply:
The number is taken from the elements in the `EcalTimeBiasCorrections_data2011` tag used for reconstruction. The MC tag has fewer elements in fact. @valsdav can perhaps comment on why it was fixed

Reply:
Naively making it a `std::vector` instead leads to compilation errors of the actual algorithm code.
```
CMSSW_13_3_0_pre4/src/RecoLocalCalo/EcalRecProducers/plugins/alpaka/TimeComputationKernels.h:994:99: error: reference to __host__ function 'data' in __host__ __device__ function
   auto* const amplitudeBins = isBarrel ? conditionsDev.timeBiasCorrections_amplitude_EB().data()
   ```

Reply:
You definitely cannot use an `std::vector` in a SoA.

If it is a _requirement_ that these collections can have a dynamic size, that can probably be arranged, but it will require significant work.

In that case, I would ask to clarify how stringent a requirement that is (_e.g._ is it enforced for all existing conditions?), and if it would be acceptable to proceed with the current approach, while we implement the necessary functionality.

Reply:
I think for the moment these can stay with a fixed size.
The time bias corrections have not been changed since 2011. I will add some code to the ESProducer to make sure that if there are fewer than the maximum values (like for MC) the extra ones have some sane values as well.
Also the fit parameters are unlikely to change in the foreseeable future.

Reply:
I added another two scalars that hold the actual size for EB and EE (like in the CUDA version) to make sure that only the valid elements are accessed. The number of bins in the constants are now the maximum number of elements.